metadata/metadata.c

```diff
@@ -470,18 +540,18 @@ VirtualDisk *create_virtual_disk(Metadata *metadata, char *name, uint64_t size,
 
     if (metadata != NULL)
     {
-        _add_virtual_disk(metadata, virtual_disk);
+        _add_virtual_disk(hwmetadata, virtual_disk);
     }
 
     return virtual_disk;
 }
 
-VirtualDiskRange **translate_vdaddress_to_vdranges(Allocator *allocator, VirtualDisk *vdisk, uint64_t start_address, uint64_t io_size)
+VirtualDiskRange **translate_vdaddress_to_vdranges(HyperwarpMetadata *metadata, VirtualDisk *vdisk, uint64_t start_address, uint64_t io_size)
 {
     uint64_t end_address = start_address + io_size - 1;
 
-    size_t start_index = floor(start_address / allocator->virtual_disk_range_sector_count);
-    size_t end_index = ceil(end_address / allocator->virtual_disk_range_sector_count);
+    size_t start_index = floor(start_address / metadata->allocator->virtual_disk_range_sector_count);
+    size_t end_index = ceil(end_address / metadata->allocator->virtual_disk_range_sector_count);
 
```

Comment:
this change is also not strictly required yet but it might be relevant in the future so I think that one is ok for now